### PR TITLE
[odo] update nodejs references to latest versions

### PR DESCRIPTION
**Pull request by yselkowitz**
RHOAR Node.js images are no longer supported, and RHSCL Node.js 8 is deprecated.

CC: @lholmquist

### CODE REVIEW

**Review by lholmquist — APPROVED** [review]

**Comment by kalexand-rh** [comment]
@vikram-redhat, who should review this?

**Comment by vikram-redhat** [comment]
@Preeticp PTAL

**Comment by Preeticp** [comment]
Thank you @vikram-redhat will follow up on this.

**Comment by Preeticp** [comment]
@boczkowska can you or Amit please verify this and provide your comments?

File: cli_reference/openshift_developer_cli/understanding-odo.adoc
Comment (amitkrout) [inline cli_reference/openshift_developer_cli/understanding-odo.adoc]: We are still using the name space ```rhoar-nodejs``` in our actual odo code - https://github.com/openshift/odo/blob/425d77670264efc502e02aed28a5665546bfefd8/pkg/catalog/catalog.go#L282-L290. Can you please share the supported reference to this change.

@kadel If this is the case then we need to update the image registry in our odo code base too. PTAL

Comment (lholmquist) [reply cli_reference/openshift_developer_cli/understanding-odo.adoc]: This was the blog post that sort of talks about the 1 node.js from redhat,  https://developers.redhat.com/blog/2020/01/07/red-hat-support-for-node-js/

I noticed in that code that there are also references to bucharestgold/node images.  We've actually changed the org name to nodeshift,  but we've actually replaced those community images with the ubi based ones from rhscl.   looks like that should be another PR :)

Comment (lholmquist) [reply cli_reference/openshift_developer_cli/understanding-odo.adoc]: Also,  related to my last comment,  Node 8 went EOL at the end of 2019,  so those references should be removed and Node 12 can be added

**Comment by yselkowitz** [comment]
/hold
This requires similar changes in odo itself, per earlier comments.

**Comment by girishramnani** [comment]
I tried the odo catalog command on `3.11` openshift cluster and that still has nodejs 8 image stream. We are still supporting 3.11` cluster in odo and so I woulds suggest to **not** remove node 8 support from the documentation.

**Comment by girishramnani** [comment]
opened an issue to track the task required on odo side https://github.com/openshift/odo/issues/2603

**Comment by yselkowitz** [comment]
> I tried the odo catalog command on `3.11` openshift cluster and that still has nodejs 8 image stream. We are still supporting 3.11` cluster in odo and so I would suggest to **not** remove node 8 support from the documentation.

nodejs-8-rhel7 as well as both versions of rhoar-nodejs are no longer supported.  While OpenShift does not update imagestreams in stable branches, that does not mean we should encourage further use of unsupported versions.

**Pull request closed without merging**
